src/map/cMap.cpp

```diff
@@ -98,6 +105,7 @@ void cMap::init(int width, int height)
 
     this->width = width;
     this->height = height;
+    mapGeometry->resize(this->width,this->height);
```

Comment:
Thinking about this; this is a more of a set of handy calculations based on width/height of a 2d plane. We could also re-create the `mapGeometry` class, if we want to stick to more Functional Programming principles. The upside is that we don't need to worry about changing state. The downside is the (re)creation of the MapGeometry class.